src/models/input.ts

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+type optionsType = {
+  data?: Buffer | null,
+};
+
+const defaultOptions = {
+  data: null
+}
+
+class Input {
+  // Hash of the transaction is being spent
+  hash: string;
+  // Index of the outputs array from the output being spent
+  index: number;
+  // Input signed data
```

Comment:
Not necessarily "signed data". Maybe change to redeemScript?

Reply:
Just made clear that it's a signed data for P2PKH and redeemScript for P2SH